### PR TITLE
Add support for sns_topic

### PR DESCRIPTION
**Pull request by moadibfr**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | yes
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #143
| ❓ Documentation  | yes

## Description

add support for sns_topic
New repository layer for accessing sdk: allows to share code between suppliers (ans will allows to cache response)

### CODE REVIEW

**Review by eliecharra — COMMENTED** [review]
I'm totally fan of your KISS approach of aws sdk wrapping using the notion of `Client` 👍🏻 

It will also fix our problem with mock for the `ListSomethingPages` method called multiples times in test 🎉 

Will do a more precise review when PR will be ready 👍🏻

File: pkg/remote/aws/repository/sns_repository.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sns"
+	"github.com/aws/aws-sdk-go/service/sns/snsiface"
+)
+
+type SNSRepository interface {
+	ListAllTopics() ([]*sns.Topic, error)
+}
+
+type snsRepositoryImpl struct {
```
Comment (eliecharra) [inline pkg/remote/aws/repository/sns_repository.go:13]: Is the `Impl` suffix required there ?

**Review by eliecharra — CHANGES_REQUESTED** [review]
Awesome 👍🏻 

You just need to update the list of supported ressources from the doc

File: pkg/remote/aws/repository/sns_repository.go (hunk 1 above)
Comment (moadibfr) [reply pkg/remote/aws/repository/sns_repository.go:13]: No you're right it survived a refacto :joy:

**Review by eliecharra — APPROVED** [review]

**Review by wbeuil — CHANGES_REQUESTED** [review]
Can you add an acceptance test and that would be good to me :)

**Review by wbeuil — APPROVED** [review]
LGTM

**Comment by eliecharra** [comment]
Rebase needed there

**Review by wbeuil — APPROVED** [review]

**Pull request merged**